src/rosdep2/platforms/arch.py

```diff
@@ -57,8 +57,14 @@ def __init__(self):
 
     def get_install_command(self, resolved, interactive=True, reinstall=False, quiet=False):
         #TODO: interactive switch
-        packages = self.get_packages_to_install(resolved, reinstall=reinstall)        
+        packages = self.get_packages_to_install(resolved, reinstall=reinstall)
         if not packages:
             return []
-        else:
-            return [self.elevate_priv(['pacman', '-Sy', '--needed', p]) for p in packages]
+
+        command  = ['pacman', '-S']
+
+        if not interactive: command.append('--noconfirm')
+        if not reinstall:   command.append('--needed')
+        if quiet:           command.append('-q')
```

Comment:
nitpick: this repository's files do not follow PEP8, but with additions we can try to follow it where it is not at odds with the files existing style. This isn't PEP8, please put the contents of the if block on new lines.